src/plugins/discover/public/embeddable/initialize_fetch.ts

```diff
@@ -0,0 +1,229 @@
+              ? createEsqlDataSource()
+              : dataView.id
+              ? createDataViewDataSource({ dataViewId: dataView.id })
+              : undefined,
+            dataView,
+            query: searchSourceQuery,
+          });
+
+          const esqlMode = isEsqlMode(savedSearch);
+          if (
+            esqlMode &&
+            searchSourceQuery &&
+            (!fetchContext.query || isOfQueryType(fetchContext.query))
+          ) {
+            // Request ES|QL data
+            const result = await fetchEsql({
+              query: searchSourceQuery,
+              inputTimeRange: getTimeRangeFromFetchContext(fetchContext),
+              inputQuery: fetchContext.query,
+              filters: fetchContext.filters,
+              dataView,
+              abortSignal: currentAbortController.signal,
+              inspectorAdapters: discoverServices.inspector,
+              data: discoverServices.data,
+              expressions: discoverServices.expressions,
+              profilesManager: discoverServices.profilesManager,
+            });
+            return {
+              columnsMeta: result.esqlQueryColumns
+                ? getTextBasedColumnsMeta(result.esqlQueryColumns)
+                : undefined,
+              rows: result.records,
+              hitCount: result.records.length,
+              fetchContext,
+            };
+          }
+
+          const executionContext = await getExecutionContext(api, discoverServices);
+
+          /**
+           * Fetch via saved search
+           */
+          const { rawResponse: resp } = await lastValueFrom(
+            savedSearch.searchSource.fetch$({
+              abortSignal: currentAbortController.signal,
+              sessionId: searchSessionId,
+              inspector: {
+                adapter: requestAdapter,
+                title: i18n.translate('discover.embeddable.inspectorRequestDataTitle', {
+                  defaultMessage: 'Data',
+                }),
+                description: i18n.translate('discover.embeddable.inspectorRequestDescription', {
+                  defaultMessage:
+                    'This request queries Elasticsearch to fetch the data for the search.',
+                }),
+              },
+              executionContext,
+              disableWarningToasts: true,
+            })
+          );
+          const interceptedWarnings: SearchResponseWarning[] = [];
+          discoverServices.data.search.showWarnings(requestAdapter, (warning) => {
+            interceptedWarnings.push(warning);
+            return true; // suppress the default behaviour
+          });
+
+          return {
+            warnings: interceptedWarnings,
+            rows: resp.hits.hits.map((hit) => buildDataTableRecord(hit as EsHitRecord, dataView)),
+            hitCount: resp.hits.total as number,
+            fetchContext,
+          };
+        } catch (error) {
+          return { error };
```

Comment:
In investigating this, I realized my original logic was wrong - because of where I was aborting, we were hitting the `error` catch **waaay** more often than we should have been. Cleaned this up in 9e24c2d27c340074d4444ebfcd5faef6f474f4f6.

That being said, the subscribe callback isn't called because the `switchMap` is cancelling the previous observable and subscribing to a new one when another request is fired before the `await` attached to fetching the data is complete.